### PR TITLE
Support for Pattern transformation.

### LINKED ISSUE #2108: CGPattern CTM rotation should not affect the brush
If a rotation is applied to the context, it should not affect the patterns being drawn.
Patterns should always be of orientation 0.

### PR DESCRIPTION
**Pull request by msft-Jeyaram**
fixes #2108

### CODE REVIEW

File: Frameworks/CoreGraphics/CGContext.mm [hunk 1]
```diff
@@ -2537,6 +2533,18 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
...
+        // pre transform the geometry
+        ComPtr<ID2D1TransformedGeometry> transformedGeometry;
+        FAIL_FAST_IF_FAILED(
+            Factory()->CreateTransformedGeometry(geometry.Get(),
+                                                 __CGAffineTransformToD2D_F(CGContextGetUserSpaceToDeviceSpaceTransform(this)),
+                                                 &transformedGeometry));
+
+        // Draw it in device space, so transformation is not applied to the render target.
+        coordinateMode = _kCGCoordinateModeDeviceSpace;
+        geometry.Attach(transformedGeometry.Detach());
```
Comment (msft-Jeyaram) [inline Frameworks/CoreGraphics/CGContext.mm]: note: As per D2D DrawGeo.. a transformed geometry will have a thinner line width, than an untransformed geometry + transformation.

Working on a fix in the coming iteration.

**Comment by DHowett-MSFT** [comment]
Do we have any tests for _stroking_ with a pattern brush??

**Comment by msft-Jeyaram** [comment]
@DHowett-MSFT  TEST_P(CGPatternColoredRectBasedStroke, PatternStrokeRegion) {
Cover the stroke

File: tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp [hunk 2]
```diff
@@ -365,21 +365,113 @@ DISABLED_DRAW_TEST_F(CGPatternTests, PatternDrawPath, UIKitMimicTest<>) {
...
+DRAW_TEST_F(CGPatternTests, PatternFillTransformationWindowsLogoRotate, UIKitMimicTest<>) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+    CGContextRotateCTM(context, 0.4);
+    _SetPatternForFill(context, CGRectMake(0, 0, 100, 100), 100, 100, drawCoolStar, 1);
+
+    CGContextFillRect(context, bounds);
+}
+
+DRAW_TEST_F(CGPatternTests, PatternFillTransformationWindowsLogoTranslate, UIKitMimicTest<>) {
```
Comment (DHowett-MSFT) [inline tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp]: these can be parameterized on CGAffineTransform

Comment (DHowett-MSFT) [inline tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp]: also, we don't know what happens to stroke pattern brushes under transformation

File: Frameworks/CoreGraphics/CGContext.mm [hunk 3]
```diff
@@ -144,15 +143,23 @@ inline void ComputeStrokeStyle(ID2D1DeviceContext* deviceContext) {
         ComPtr<ID2D1Factory> factory;
         deviceContext->GetFactory(&factory);
 
+        ComPtr<ID2D1Factory1> factory1;
```
Comment (rajsesh-msft) [inline Frameworks/CoreGraphics/CGContext.mm]: >factory1 [](start = 30, length = 8)

factory.As(&factory1)? also no need for fail fasts here, we will crash right afterwards anyway.

**Pull request merged**